core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -2122,6 +2123,28 @@ private boolean hasNested(
       return false;
     }
 
+    /** Recursively looks for a node matching the operandPredicate. */
```

Comment:
this is usually done using a visitor patter, see SqlShuttle.

Reply:
ahh thanks for the feedback @mihaibudiu. I just pushed a refactor to use an SqlShuttle at #
226b7d3